renderer/reducers/activity.js

```diff
@@ -289,15 +297,81 @@ export const fetchActivityHistory = () => dispatch => {
     dispatch(fetchDescribeNetwork())
     dispatch(fetchChannels())
     dispatch(fetchBalance())
-    dispatch(fetchPayments())
-    dispatch(fetchInvoices())
-    dispatch(fetchTransactions())
+    dispatch(loadNextPage())
     dispatch({ type: FETCH_ACTIVITY_HISTORY_SUCCESS })
   } catch (error) {
     dispatch({ type: FETCH_ACTIVITY_HISTORY_FAILURE, error })
   }
 }
 
+export const resetActivity = () => () => {
+  paginator = null
+}
+
+/**
+ * createActivityPaginator - Creates  activity paginator object.
+ *
+ * @returns {Function} Paginator
+ */
+function createActivityPaginator() {
+  const fetchInvoices = async (pageSize, offset) => {
+    const { invoices, last_index_offset } = await grpc.services.Lightning.listInvoices({
+      num_max_invoices: pageSize,
+      index_offset: offset,
+    })
+    return { items: invoices, offset: last_index_offset }
+  }
+
+  const fetchPayments = async () => {
+    const { payments } = await grpc.services.Lightning.listPayments()
+    return { items: payments, offset: 0 }
+  }
+
+  const fetchTransactions = async () => {
+    const { transactions } = await grpc.services.Lightning.getTransactions()
+    return { items: transactions, offset: 0 }
+  }
+
+  return combinePaginators(fetchInvoices, fetchPayments, fetchTransactions)
+}
+
+/**
+ * getPaginator - Returns current activity paginator object. This acts as a singleton
+ * and creates paginator if it's not initialized.
+ *
+ * @returns {Function} Paginator
+ */
+function getPaginator() {
+  if (!paginator) {
+    paginator = createActivityPaginator()
+  }
+  return paginator
+}
+
+// Loads next activity page if it's avaialable
+export const loadNextPage = () => async (dispatch, getState) => {
+  const paginator = getPaginator()
+  if (getState().activity.hasNextPage) {
+    const { items, hasNextPage } = await paginator(25)
```

Comment:
Lets extract hard coded `25` either to a const at the top of this file and/or to a setting on the global config file.